### PR TITLE
Text prefab with optimized dimension and rendering quality

### PR DESCRIPTION
**Pull request by cre8ivepark**
Text is the one of the most important component in Holographic apps. To display text, there are two types of text components you can use in Unity — UI Text and 3D Text Mesh. However, there are two problems with these.
**1. They are blurry in default. You need to tweak some of the variables to get the sharp high-quality text.**
**2. They are too big in default. You need to scale down a lot to get manageable size in HoloLens.**

These Text prefabs provide better text rendering quality and proper dimension. Below describes the way to find the scaling factor to get proper dimension with UI Text and 3D Text Mesh. 

# Working with Fonts in Unity
Unity assumes all new elements added to a scene is 1 Unity Unit in size, or 100% Transform scale, which translates to about 1 meter on the HoloLens. In the case of fonts, the bounding box for a 3D TextMesh comes in, by default at about 1 meter in height.

# Font Scale and Font Sizes
Most visual designers use Points to define font sizes in the real world, as well as their design programs. There are about 2835 (2,834.645666399962) points in 1 meter. Based on the point system conversion to 1 meter and Unity's default TextMesh Font Size of 13, the simple math of 13 divided by 2835 equals 0.0046 (0.004586111116 to be exact) provides a good standard scale to start with, though some may wish to round to 0.005.

Either way, scaling the Text object or container to these values will not only allow for the 1:1 conversion of font sizes from a design program, but also provides a standard to maintain consistency throughout the application or game.

# UI Text
When adding a UI or canvas based Text element to a scene, the size disparity is greater still. The differences in the two sizes is about 1000%, which would bring the scale factor for UI based Text components to 0.00046 (0.0004586111116 to be exact) or 0.0005 for the rounded value.
Disclaimer: The default value of any font may be effected by the texture size of that font or how the font was imported into Unity. These tests were performed based on the default Arial font in Unity, as well as one other imported font.

Based on these scaling factors, we have created two prefabs - UI Text and 3D Text Mesh. Developers can use these prefabs to get sharp text and consistent font size. Since default font material does not support occlusion, we also want to include simple shader that supports occlusion.

## Blurry default text - UI Text and 3D Text Mesh
unitytexts

## Sharp text rendering quality with proper dimension
2016-11-06 00_19_44-unity personal 64bit - untitled - textquality - windows store apps_ _dx11_

## Text examples with the prefabs
textprefabinstructions05

## Default font shader vs shader with occlusion support(included in this update)
textocclusion2

### CODE REVIEW

**Comment by jwittner** [comment]
This is awesome! I see you added some prefabs to the examples, that's great!  Can you add a test scene that includes all this new functionality so we've got one nice place to see it all and going forward verify that nothing has been broken?

**Comment by cre8ivepark** [comment]
Thanks @jwittner for the suggestion! I have added test scene with sample layouts.
2016-11-22 16_22_59-unity personal 64bit - textprefabsamplestest unity - textprefabtest5 - pc mac

File: Assets/HoloToolkit/UI/Shaders/3DTextShader.shader [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿Shader "Custom/3DTextShader"{
```
Comment (NeerajW) [inline Assets/HoloToolkit/UI/Shaders/3DTextShader.shader]: So we are consistent with other HoloToolkit shaders, please make this:

Shader "HoloToolkit/3DTextShader"

Comment (NeerajW) [inline Assets/HoloToolkit/UI/Shaders/3DTextShader.shader]: Or something inside the HoloToolkit drop down for ease of discovery in Unity.

**Comment by cre8ivepark** [comment]
@NeerajW Thanks Neeraj, updated to 'HoloToolkit/3DTextShader'. Also updated related instructions image file.

**Pull request merged**